src/main/java/com/tstepnik/planner/controller/RegistrationController.java

```diff
@@ -17,19 +19,21 @@
 public class RegistrationController {
 
     private final RegistrationService registrationService;
+    private final RegisterUserMapper mapper;
 
     @Autowired
-    public RegistrationController(RegistrationService registrationService) {
+    public RegistrationController(RegistrationService registrationService, RegisterUserMapper mapper) {
         this.registrationService = registrationService;
+        this.mapper = mapper;
     }
 
     @PostMapping("/register")
-    public ResponseEntity<User> registerUser(@RequestBody User user) {
-        Optional<User> registerUser = Optional.ofNullable(registrationService.register(user));
+    public ResponseEntity<RegisterUserDto> registerUser(@RequestBody RegisterUserDto user) {
+        Optional<User> registerUser = Optional.ofNullable(registrationService.register(mapper.toUser(user)));
         if (registerUser.isEmpty()) {
             //TODO remove it once you add exception handler.
             return new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
         }
-        return new ResponseEntity<>(registerUser.get(), HttpStatus.CREATED);
+        return new ResponseEntity<>(mapper.toDto(registerUser.get()), HttpStatus.CREATED);
```

Comment:
Nonetheless, accepting `RegisterUserDTO` makes sense, since we need info like password, username, etc, but returning it doesn't make sense. Are we returning password? I would just return User/Userdto or nothing at all.